tests/test_serialization.py

```diff
@@ -148,6 +151,35 @@ def test_pickle(obj: Any, ref_type: Any) -> None:
             assert c1._metadata.key_type is Any
 
 
+@pytest.mark.parametrize(  # type: ignore
+    "obj,ref_type,key",
+    [
+        (OmegaConf.structured(DictUnion), Union[int, float], "dict"),
+        (OmegaConf.structured(ListUnion), Union[int, float], "list"),
+        (OmegaConf.structured(UnionClass), Union[int, str], "foo"),
+    ],
+)
+def test_pickle_union(obj: Any, ref_type: Any, key: str) -> None:
+    import collections
+
+    from omegaconf._utils import get_union_types
+
+    with tempfile.TemporaryFile() as fp:
```

Comment:
You could use `pickle.loads` and `pickle.dumps` to avoid having to mess with files.

Reply:
nice, we should probably clean up the pickle serialization tests.